src/market/directory.rs

```diff
@@ -138,7 +160,7 @@ fn write_default_directory_config(config_path: &PathBuf) -> std::io::Result<()>
     write_default_config(config_path, config_string)
 }
 
-pub fn start_directory_server(directory: Arc<DirectoryServer>) {
+pub fn start_directory_server(directory: Arc<DirectoryServer>) -> Result<(), DirectoryServerError> {
     let address_file = directory.data_dir.join("addresses.dat");
 
     let addresses = Arc::new(RwLock::new(HashSet::new()));
```

Comment:
This will break our current implementation. We need to modify the logic for GET to either send the addresses in reverse order of arrival or add a custom comparator to the hash set to order the elements based on their arrival. If we don't implement this, it may allow the client to ping the maker with an inactive URL. 

The primary reason for this PR is to change the GET logic to utilize the order of arrival for correctly organizing the addresses. Alternatively, we could modify the client logic to ping the addresses, and if one doesn't respond, it can request a new one from the directory server.

Reply:
> We need to modify the logic for GET to either send the addresses in reverse order of arrival or add a custom comparator

I didn't think of the ordering. The ordering logic could be done on the taker side too. Or can be added here also. Ack on both of them and should be part of scope of this PR. 

Its not a breaking bug currently, but ya unsorted list at the Taker's address book layer is not good idea. 

> The primary reason for this PR is to change the GET logic to utilize the order of arrival for correctly organizing the addresses. Alternatively, we could modify the client logic to ping the addresses, and if one doesn't respond, it can request a new one from the directory server.

This sounds overly complicated to me. Why have two sources of addresses when we can have just one hashset wrapped in Arc<Mutex> and use that as the source-of-truth. Much simpler code to write and maintain, less bug surface. (I feel Mutex will be more appropriate here than RWLocks?)